src/lib/kafka_storage.ts

```diff
@@ -1,59 +1,59 @@
-const crypto = require('crypto');
-const Kafka = require('node-rdkafka');
-const { BLOCKCHAIN } = require('./constants');
-const ZookeeperClientAsync = require('./zookeeper_client_async');
-const { logger, SYSLOG_LOG_LEVEL, log_according_to_syslog_level } = require('./logger');
-
-const ZOOKEEPER_URL = process.env.ZOOKEEPER_URL || 'localhost:2181';
-const ZOOKEEPER_RETRIES = parseInt(process.env.ZOOKEEPER_RETRIES || '0');
-const ZOOKEEPER_SPIN_DELAY = parseInt(process.env.ZOOKEEPER_SPIN_DELAY || '1000');
-const ZOOKEEPER_SESSION_TIMEOUT = parseInt(process.env.ZOOKEEPER_SESSION_TIMEOUT || '30000');
-
-const FORMAT_HEADER = 'format=json;';
-const RDKAFKA_DEBUG = process.env.RDKAFKA_DEBUG || null;
-const KAFKA_URL = process.env.KAFKA_URL || 'localhost:9092';
-const KAFKA_COMPRESSION_CODEC = process.env.KAFKA_COMPRESSION_CODEC || 'lz4';
-const KAFKA_FLUSH_TIMEOUT = parseInt(process.env.KAFKA_FLUSH_TIMEOUT || '10000');
-const BUFFERING_MAX_MESSAGES = parseInt(process.env.BUFFERING_MAX_MESSAGES || '150000');
-const TRANSACTIONS_TIMEOUT_MS = parseInt(process.env.TRANSACTIONS_TIMEOUT_MS || '60000');
-const KAFKA_MESSAGE_MAX_BYTES = parseInt(process.env.KAFKA_MESSAGE_MAX_BYTES || '10485760');
-
-process.on('unhandledRejection', (reason, p) => {
+import crypto from 'crypto';
+import Kafka, { LibrdKafkaError, ProducerGlobalConfig } from 'node-rdkafka';
+import { BLOCKCHAIN } from './constants';
+import ZookeeperClientAsync from './zookeeper_client_async';
+import { log_according_to_syslog_level, logger, SYSLOG_LOG_LEVEL } from './logger';
+
+
+const ZOOKEEPER_URL: string = process.env.ZOOKEEPER_URL || 'localhost:2181';
+const ZOOKEEPER_RETRIES: number = parseInt(process.env.ZOOKEEPER_RETRIES || '0');
+const ZOOKEEPER_SPIN_DELAY: number = parseInt(process.env.ZOOKEEPER_SPIN_DELAY || '1000');
+const ZOOKEEPER_SESSION_TIMEOUT: number = parseInt(process.env.ZOOKEEPER_SESSION_TIMEOUT || '30000');
+
+const FORMAT_HEADER: string = 'format=json;';
+const RDKAFKA_DEBUG: string | null = process.env.RDKAFKA_DEBUG || null;
+const KAFKA_URL: string = process.env.KAFKA_URL || 'localhost:9092';
+const KAFKA_COMPRESSION_CODEC: string = process.env.KAFKA_COMPRESSION_CODEC || 'lz4';
+const KAFKA_FLUSH_TIMEOUT: number = parseInt(process.env.KAFKA_FLUSH_TIMEOUT || '10000');
+const BUFFERING_MAX_MESSAGES: number = parseInt(process.env.BUFFERING_MAX_MESSAGES || '150000');
+const TRANSACTIONS_TIMEOUT_MS: number = parseInt(process.env.TRANSACTIONS_TIMEOUT_MS || '60000');
+const KAFKA_MESSAGE_MAX_BYTES: number = parseInt(process.env.KAFKA_MESSAGE_MAX_BYTES || '10485760');
```

Comment:
I'm not sure that explicit type definitions are required here if this type is private (you don't export it)